### DISCUSSION TITLE
Usage with react-virtualized InfiniteLoader

### DISCUSSION THREAD

**Discussion by mateja176** [TanStack/query#829] [General]
I've been trying to make the integration between `react-query` and `react-virtualized`'s `InfiniteLoader` possible, however I got stuck at a particular point.

Namely, the `InfinteLoader`'s `fetchMore` function is invoked with an `IndexRange` (`{ startIndex: number, stopIndex: number }`). This is important because the batch size can vary and usually the initially batch size is larger than the subsequent batches. Hence it is imperative for the `queryFn` to have access to the `startIndex`, `stopIndex` or a computed `limit`. For this integration I've been using the firebase sdk however the principal of a cursor is used more widely. I won't go into the specifics, however, `firebase`'s `startAfter` collection filter is accepts the reference to the last loaded document. Fortunately, the `getFetchMore` function is invoked with the last batch and the last document in the batch represents the cursor after which the next batch starts. The point being that there is not way to pass both the `limit` and the last document reference to the query function in order to load the new batch.

Is there are workaround?

**Comment by sazzouz** [comment]
Did you make any progress with this? I am looking for something similar
